internal/controllers/extension_controller.go

```diff
@@ -228,28 +248,37 @@ func (r *ExtensionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// TODO: follow up with mapping of all the available App statuses: https://github.com/carvel-dev/kapp-controller/blob/855063edee53315811a13ee8d5df1431ba258ede/pkg/apis/kappctrl/v1alpha1/status.go#L28-L35
-// mapAppStatusToInstalledCondition currently maps only the installed condition.
-func mapAppStatusToInstalledCondition(existingApp *kappctrlv1alpha1.App, ext *ocv1alpha1.Extension, bundle *catalogmetadata.Bundle) {
-	appReady := findStatusCondition(existingApp.Status.GenericStatus.Conditions, kappctrlv1alpha1.ReconcileSucceeded)
-	if appReady == nil {
-		ext.Status.InstalledBundle = nil
-		setInstalledStatusConditionUnknown(&ext.Status.Conditions, "install status unknown", ext.Generation)
+// mapAppStatusToCondition maps the reconciling/deleting App conditions to the installed/deleting conditions on the Extension.
+func mapAppStatusToCondition(existingApp *kappctrlv1alpha1.App, ext *ocv1alpha1.Extension) {
+	// Note: App.Status.Inspect errors are never surfaced to App conditions, so are currently ignored when determining App status.
+	if ext == nil || existingApp == nil {
 		return
 	}
-
-	if appReady.Status != corev1.ConditionTrue {
-		ext.Status.InstalledBundle = nil
-		setInstalledStatusConditionFailed(
-			&ext.Status.Conditions,
-			appReady.Message,
-			ext.GetGeneration(),
-		)
-		return
+	message := existingApp.Status.FriendlyDescription
+	if len(message) == 0 || strings.Contains(message, "Error (see .status.usefulErrorMessage for details)") {
+		message = existingApp.Status.UsefulErrorMessage
 	}
 
-	ext.Status.InstalledBundle = bundleMetadataFor(bundle)
-	setInstalledStatusConditionSuccess(&ext.Status.Conditions, appReady.Message, ext.Generation)
+	appStatusMapFn := map[kappctrlv1alpha1.ConditionType]func(*[]metav1.Condition, string, int64){
+		kappctrlv1alpha1.Deleting:           setProgressingStatusConditionProgressing,
+		kappctrlv1alpha1.Reconciling:        setProgressingStatusConditionProgressing,
+		kappctrlv1alpha1.DeleteFailed:       setProgressingStatusConditionFailed,
+		kappctrlv1alpha1.ReconcileFailed:    setProgressingStatusConditionFailed,
+		kappctrlv1alpha1.ReconcileSucceeded: setProgressingStatusConditionSuccess,
+	}
+	for cond := range appStatusMapFn {
+		if c := findStatusCondition(existingApp.Status.GenericStatus.Conditions, cond); c != nil && c.Status == corev1.ConditionTrue {
+			if len(message) == 0 {
+				message = c.Message
+			}
+			appStatusMapFn[cond](&ext.Status.Conditions, fmt.Sprintf("App %s: %s", c.Type, message), ext.Generation)
+			return
+		}
+	}
+	if len(message) == 0 {
+		message = "waiting for app"
+	}
+	setProgressingStatusConditionProgressing(&ext.Status.Conditions, message, ext.Generation)
```

Comment:
I'm not a huge fan of looping through all the status conditions 5 times but since we are moving away from building on top of the App CR and this is an optimization that could be made in the future I'm fine with it for now